drivers/dome/ddw_dome.cpp

```diff
@@ -0,0 +1,560 @@
+{
+    int nbytes_read = 0, rc = -1;
+    char errstr[MAXRBUF];
+
+    // Read response and parse it
+    char response[256] = { 0 };
+
+    // Read buffer
+    if ((rc = tty_nread_section(PortFD, response, sizeof(response) - 1, '\r', DDW_TIMEOUT, &nbytes_read)) != TTY_OK)
+    {
+        tty_error_msg(rc, errstr, MAXRBUF);
+        LOGF_ERROR("Error reading: %s.", errstr);
+    }
+    status = response;
+    return rc;
+}
+
+void DDW::parseGINF(const char *response)
+{
+    /* GINF packet structure:
+
+    Field   Content     Note (each datum is separated by comma)
+    1       V#          Denotes Version Data.  E.g., V1 
+    2       Dticks      DTICKS is dome circumference in ticks 0-32767.  Value is sent as a string of characters, e.g., 457.  Leading zeros not transmitted.
+    3       Home1       Azimuth location of the HOME position in ticks 0-32767
+    4       Coast       Coast value in ticks (0-255)
+    5       ADAZ        Current dome azimuth in Ticks 0-32767
+    6       Slave       Slave Status 0=slave off 1=slave on
+    7       Shutter     Shutter status 0=indeterminate, 1=closed, 2=open
+    8       DSR status  DSR Status 0=indet, 1=closed, 2=open
+    9       Home        Home sensor 0=home, 1=not home
+    10      HTICK_CCLK  Azimuth ticks of counterclockwise edge of Home position
+    11      HTICK_CLK   Azimuth ticks of clockwise edge of Home position
+    12      UPINS       Status of all user digital output pins
+    13      WEAAGE      Age of weather info in minutes 0 to 255 (255 means expired)
+    14      WINDDIR     0-255 wind direction (use (n/255)*359 to compute actual direction), subtract dome azimuth if weather module is mounted on dome top.
+    15      WINDSPD     Windspeed 0-255 miles per hour
+    16      TEMP        Temperature 0-255, representing -100 to 155 degrees F
+    17      HUMID       Humidity 0-100% relative
+    18      WETNESS     Wetness 0 (dry) to 100 (soaking wet)
+    19      SNOW        Snow 0 (none) to 100 (sensor covered)
+    20      WIND PEAK   Windspeed Peak level over session 0-255 miles per hour
+    21      SCOPEAZ     Scope azimuth from LX-200 (999 if not available)
+    22      INTDZ       Internal “deadzone”- angular displacement around the dome opening centerline within which desired dome azimuth can change without causing dome movement.
+    23      INTOFF      Internal offset- angular distance DDW will add to the desired azimuth, causing the dome to preceed the telescope’s position when a slaved goto occurs.
+    24      car ret
+    25      car ret
+    */
+
+    int version;
+    int dticks;
+    int homepos;
+    int coast;
+    int azimuth;
+    int slave;
+    int shutter;
+    int dsr;
+    int home;
+    int htick_cclk;
+    int htick_clk;
+    int upins;
+    int weatherAge;
+    int windDir;
+    int windSpd;
+    int temperature;
+    int humidity;
+    int wetness;
+    int snow;
+    int windPeak;
+    int scopeAzimuth;
+    int deadzone;
+    int offset;
```

Comment:
best to initialize to some value? 0?

Reply:
Style-wise perhaps, but the values are not used uninitialized as the return value of sscanf is checked so that all values are initialized.